rs/replicated_state/src/metadata_state.rs

```diff
@@ -175,6 +175,10 @@ pub struct SystemMetadata {
     /// by aggregating them and storing a running total over multiple days by node id and
     /// timestamp. Observations of blockmaker stats are performed each time a batch is processed.
     pub blockmaker_metrics_time_series: BlockmakerMetricsTimeSeries,
+
+    /// Modifications to the state that have not been applied yet to the next checkpoint.
+    /// This field is transient and is emptied at the beginning of each checkpoint.
+    pub unflushed_checkpoint_ops: UnflushedCheckpointOps,
```

Comment:
> It would be weird for the canister migration logic to populate a field in CanisterSnapshots, so this refactor moves the field out to a more general location.

I'm not sure I understand why that is. At first sight, it feels weirder to have state that seems like it should be internal to `CanisterSnapshots` live next to it and be passed as mutable argument to most (?) of its methods.

Reply:
Per our offline discussion, my confusion stemmed from the fact that I missed the recasting of "snapshot operations" into more general "checkpoint operations" (even though the set of operations remains unchanged for now).

With that in mind, I'll make another pass over the PR. Thanks for the clarification.